addons/implied-repos/common/src/main/java/org/commonjava/indy/implrepo/change/ImpliedRepositoryDetector.java

```diff
@@ -459,6 +472,43 @@ else if ( !config.isIncludeSnapshotRepos() && !repo.isReleasesEnabled() )
                 else
                 {
                     logger.debug( "Found existing RemoteRepositories: {}", rrs );
+
+                    for ( final RemoteRepository rr : rrs )
+                    {
+                        rr.setMetadata( METADATA_ORIGIN, IMPLIED_REPO_ORIGIN );
+                        try
+                        {
+                            rr.setMetadata( IMPLIED_BY_STORES, mapper.writeValueAsString(
```

Comment:
I think if we're reusing an existing remote repo, we should add the current job key to any existing job keys, so we don't lose the old information (if it's already there).

Reply:
Yeah, merging here looks good.